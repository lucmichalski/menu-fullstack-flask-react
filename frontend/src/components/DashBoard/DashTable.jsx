import React from "react";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";

export default function DashTable({ pedidos, clientes }) {
  const getClient = client_id => {
    let client = clientes.find(cliente => cliente.id == client_id)
      .primeiro_nome;
    return client;
  };

  return (
    <TableContainer component={Paper}>
      <div style={{ textAlign: "center" }}>
        <b>Ãšltimos pedidos</b>
      </div>
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>
              <b>Cliente</b>
            </TableCell>
            <TableCell align="right">
              <b>Valor</b>
            </TableCell>
            <TableCell align="right">
              <b>Status</b>
            </TableCell>
            <TableCell align="right">
              <b>Data</b>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {pedidos
            .reverse()
            .slice(0, 5)
            .map(pedido => (
              <TableRow key={pedido.id}>
                <TableCell component="th" scope="row">
                  {getClient(pedido.cliente_id)}
                </TableCell>
                <TableCell align="right">{pedido.valor}</TableCell>
                <TableCell align="right">{pedido.status}</TableCell>
                <TableCell align="right">
                  {new Date(pedido.data).toLocaleDateString("pt-BR")}
                </TableCell>
              </TableRow>
            ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
