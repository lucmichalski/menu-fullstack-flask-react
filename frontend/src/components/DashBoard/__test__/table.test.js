import React from "react";
import ReactDOM from "react-dom";
import DashTable from "../DashTable";
import renderer from "react-test-renderer";

const clientes = [
  {
    id: 1,
    primeiro_nome: "test_client"
  }
];

const pedidos = [
  {
    id: 1,
    cliente_id: 1,
    valor: 100,
    status: null,
    data: null
  }
];

test("renderiza corretamente", () => {
  const tree = renderer
    .create(<DashTable clientes={clientes} pedidos={pedidos} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
