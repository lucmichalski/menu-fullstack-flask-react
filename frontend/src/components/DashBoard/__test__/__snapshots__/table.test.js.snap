// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza corretamente 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <b>
      Ãšltimos pedidos
    </b>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <b>
            Cliente
          </b>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          <b>
            Valor
          </b>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          <b>
            Status
          </b>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          <b>
            Data
          </b>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          role="cell"
          scope="row"
        >
          test_client
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          100
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        />
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          12/31/1969
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
